app/models/dialog_group.rb

```diff
@@ -38,6 +38,10 @@ def validate_children
       errors.add(:base, _("Box %{box_label} must have at least one Element") % {:box_label => label})
     end
 
+    if dialog_fields.collect(&:name).duplicates.present?
+      errors.add(:base, _("Dialog field name cannot be duplicated on a dialog: %{duplicates}") % {:duplicates => dialog_fields.collect(&:name).duplicates.join(', ')})
```

Comment:
This does not belong on the group object since `dialog_fields` would only validate objects in that group and not the entire dialog.